app/Models/Album.php

```
@@ -211,100 +180,184 @@ public function cover(): HasOne
-	 * Setter/Mutator for attribute `min_taken_at`.
+	 * Recursively deletes the album incl. potential sub-albums and photos.
+	 *
+	 * Note, this method only deletes albums and photos which are owned by
+	 * the current user.
+	 * If the album is not empty (after all sub-albums and photos which have
+	 * been owned by the user have been deleted), the album is not deleted.
 	 *
-	 * Actually, this method should be a no-op and throw an exception.
-	 * The attribute `min_taken_at` is a transient attribute of the model
-	 * and cannot be persisted to database.
-	 * It is calculated by the DB back-end upon fetching the model.
-	 * Hence, it wrong to try to set this attribute.
-	 * However, {@link AlbumCast::toTagAlbum()} does it nonetheless, so we
-	 * don't throw an exception until that method is fixed.
+	 * Note, the parameter `$skipTreeFixing` should not be used by an external
+	 * caller (but left equal to its default `false`).
+	 * This flag is only internally used by this method for better performance
+	 * and skips rebuilding the tree after each recursion step.
 	 *
-	 * TODO: Fix {@link AlbumCast::toTagAlbum()}.
+	 * @param bool $skipTreeFixing
 	 *
-	 * @param Carbon|null $value
+	 * @return bool
 	 */
-	protected function setMinTakenAtAttribute(?Carbon $value): void
+	public function delete(bool $skipTreeFixing = false): bool
 	{
-		// Uncomment the following line, after AlbumCast::toTagAlbum() has been fixed
-		//throw new \BadMethodCallException('Attribute "min_taken_at" must not be set as it is a virtual attribute');
+		$success = true;
+
+		$photos = $this->photos()
+			->where('owner_id', '=', AccessControl::id())
+			->get();
+		/** @var Photo $photo */
+		foreach ($photos as $photo) {
+			// This also takes care of proper deletion of physical files from disk
+			$success &= $photo->delete();
+		}
+
+		$albums = $this->children()
+			->whereHas(
+				'base_class',
+				fn (Builder $q) => $q->where('owner_id', '=', AccessControl::id())
+			)
+			->get();
```

Again, what's the purpose of these ownership tests? Right now they should not be needed. Are you planning to introduce mixed-ownership albums?

---

I was not planning it, I thought it was already there. Probably my mistake and I am too lazy to check the old code. I thought these checks were there, too.

Of course, we can through that away if it not needed. But we could also keep it to avoid surprises in the future, if someone introduces such a feature.

---

As before, I see a value of having some sanity checks, but perhaps again it would be better to throw an exception if they trigger, so that it's pretty clear what the intention is?

And yeah, we do regularly get requests to add support for albums that multiple users could upload to (basically, a read-write version of our current read-only shared albums, I guess). I'm not against such a feature but I think I would still prefer if any uploaded photos became owned by the album owner.

---

They are gone now. My fault to assume we had something like mixed ownership.